### PR TITLE
chore(parameters): Use AppConfig L2 constructs for integration tests

### LINKED ISSUE #2396: Maintenance: adopt AppConfig L2 constructs in Parameters integration tests
### Summary

AWS CDK has added [L2 constructs for AWS AppConfig](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_appconfig-readme.html) that allow to create AppConfig resources more easily.

The AppConfig Provider part of the Parameters utility uses L1 constructs that are more verbose and require explicit management of dependencies between entities.

We should replace the existing construct with the newer.

### Why is this needed?

So that the integration tests for the Parameters utility can become a little less verbose and less brittle.

### Which area does this relate to?

Tests

### Solution

Replace the code [found here](https://github.com/aws-powertools/powertools-lambda-typescript/blob/2a69d41157a3c51055f1e2a30a95650c4500d5d1/packages/parameters/tests/helpers/resources.ts#L243-L311) with the L2 constructs.

### Acknowledgment

- [X] This request meets [Powertools for AWS Lambda (TypeScript) Tenets](https://docs.powertools.aws.dev/lambda/typescript/latest/#tenets)
- [ ] Should this be considered in other Powertools for AWS Lambda languages? i.e. [Python](https://github.com/aws-powertools/powertools-lambda-python/), [Java](https://github.com/aws-powertools/powertools-lambda-java/), and [.NET](https://github.com/aws-powertools/powertools-lambda-dotnet/)

### Future readers

Please react with 👍 and your use case to help us understand customer demand.

### PR DESCRIPTION
**Pull request by daschaa**
### Changes

This PR implements the L2 constructs for the integration tests in the parameters package. Closes #2396

**Issue number:** #2396

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Comment by dreamorosi** [comment]
Preliminary run for the integration tests is green: https://github.com/aws-powertools/powertools-lambda-typescript/actions/runs/9086242800

Whenever you want me to run it again, just tag me and I'll do it (it's manual for PR coming from forks).

**Comment by dreamorosi** [comment]
Like before, if you could leave a comment under the original issue I'll assign it to you - so that everyone knows the issue is taken.

**Comment by daschaa** [comment]
@dreamorosi Thanks for your fast feedback, I really appreciate it! 

I just marked this PR as a draft because I hadn't got the time to run the integration tests, I just compared the synthed output and tried to remodel the current behavior. 

So did you let the integration tests run with the current state of this PR? Can I mark it as "Ready for Review" already?

**Comment by dreamorosi** [comment]
Yes, I ran it with the latest commit around the time I left my comment.

You can mark it as ready whenever you're ready - no rush from my side.

File: packages/parameters/tests/helpers/resources.ts [hunk 1]
```diff
@@ -349,7 +329,7 @@ class TestAppConfigWithProfiles extends Construct {
     this.profiles.forEach((profile) => {
       const appConfigConfigurationArn = Stack.of(fn).formatArn({
         service: 'appconfig',
-        resource: `application/${profile.applicationId}/environment/${profile.environmentId}/configuration/${profile.configurationProfileId}`,
+        resource: `application/${profile.application.applicationId!}/environment/${profile.deployTo![0].environmentId}/configuration/${profile.configurationProfileId}`,
```
Comment (dreamorosi) [inline packages/parameters/tests/helpers/resources.ts:332]: Is this non-null assertion needed?

I haven't checked in the IDE (can do tomorrow), but the [automated scanning is flagging it](https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&pullRequest=2524&id=aws-powertools_powertools-lambda-typescript&open=AY94zAtTdMQlEnSzB59V) as unnecessary.

If it's a false positive let me know and I'll mark it as such.

Comment (daschaa) [reply packages/parameters/tests/helpers/resources.ts:332]: I think I added it because my IDE was complaining. But I‘ll check it again.

Comment (dreamorosi) [reply packages/parameters/tests/helpers/resources.ts:332]:
```suggestion
        resource: `application/${profile.application.applicationId}/environment/${profile.deployTo![0].environmentId}/configuration/${profile.configurationProfileId}`,
```

Comment (dreamorosi) [reply packages/parameters/tests/helpers/resources.ts:332]: Ok - I checked the code & the docs. The `applicationId` property is always defined and it's a string, so it's safe to remove the non-null assertion.

The other one on `deployTo` is valid, so we can keep it.

**Review by dreamorosi — CHANGES_REQUESTED** [review]
Left one minor comment [here](https://github.com/aws-powertools/powertools-lambda-typescript/pull/2524/files#r1601260480). Once it's addressed I'll run the integration tests again and if they're green I'll merge it.

Thanks again for your contribution!

**Review by dreamorosi — APPROVED** [review]
I have ran the [integration tests](https://github.com/aws-powertools/powertools-lambda-typescript/actions/runs/9099482277) once again and they're green.

Thank you so much for working on this PR!

**Pull request merged**